### PR TITLE
Allow users to specify an SSHKey from their Terraform Cloud Organization.

### PR DESCRIPTION
**Pull request by rojopolis**
Relates: #25

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates or Closes #00000000

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-k8s/blob/master/CHANGELOG.md):

```release-note
Add `sshKeyID` to the spec.  This causes the workspace to be created (or updated) with the referenced key from the users Terraform Cloud Organization.  This allows users to reference modules in private git repos.
```

Output:

```
$ kubectl describe workspace
Name:         hello
Namespace:    default
Labels:       <none>
Annotations:  kubectl.kubernetes.io/last-applied-configuration:
                {"apiVersion":"app.terraform.io/v1alpha1","kind":"Workspace","metadata":{"annotations":{},"name":"hello","namespace":"default"},"spec":{"m...
API Version:  app.terraform.io/v1alpha1
Kind:         Workspace
Metadata:
  Creation Timestamp:  2020-04-16T00:46:13Z
  Finalizers:
    finalizer.workspace.app.terraform.io
  Generation:        5
  Resource Version:  22650
  Self Link:         /apis/app.terraform.io/v1alpha1/namespaces/default/workspaces/hello
  UID:               6aedfa2a-caf3-4c1c-9ca7-0797d0cc9f22
Spec:
  Module:
    Source:      joatmon08/hello/random
    Version:     4.0.0
  Organization:  rojopolis
  Outputs:
    Key:                 my_pets
    Module Output Name:  list_of_pets
  Secrets Mount Path:    /tmp/secrets
  Ssh Key ID:            sshkey-D4fmNfDCWAoBJopq
  Variables:
    Environment Variable:  false
    Key:                   hello
    Sensitive:             false
    Value:                 rosario
    Environment Variable:  false
    Key:                   second_hello
    Sensitive:             false
    Value From:
      Config Map Key Ref:
        Key:               to
        Name:              say-hello
    Environment Variable:  false
    Key:                   secret_key
    Sensitive:             true
    Environment Variable:  true
    Key:                   AWS_SECRET_ACCESS_KEY
    Sensitive:             true
    Environment Variable:  true
    Key:                   CONFIRM_DESTROY
    Sensitive:             false
    Value:                 1
Status:
  Outputs:
    Key:         my_pets
    Value:       ["wondrous-chicken","stable-firefly"]
  Run ID:        run-VcSQEVsphX6qe1i6
  Run Status:    applied
  Workspace ID:  ws-fZ1KJ53xGNExzp3q
Events:          <none>

```

### CODE REVIEW

**Comment by rojopolis** [comment]
@joatmon08 is this project accepting PRs?

**Comment by aareet** [comment]
Hi @rojopolis, thank you for your contributions. We are certainly accepting PRs for this project, apologies for the delay in this receiving attention. It's in our queue for review.

File: operator/deploy/crds/app.terraform.io_workspaces_crd.yaml [hunk 1]
```diff
@@ -167,6 +170,7 @@ spec:
           - module
           - organization
           - secretsMountPath
+          - sshKeyID
```
[the remark this answers is not in the document]
Comment (rojopolis) [reply operator/deploy/crds/app.terraform.io_workspaces_crd.yaml:173]: > I was also hoping to get some advice regarding testing this. I can set my `sshKeyID` in the Workspace CR and apply it, but it doesn't seem to trigger an update in TFC. I have a named key that I called `testkey` and I was setting it in the CR, but I never saw it show up here in the web console.

Can you list the steps to reproduce this?  Specifically, is this a new workspace in which an ssh key was specified, or a workspace that was created and then had the CR updated to include an ssh key?

I'll take a look at the other issue... this should indeed be an optional attribute.

**Comment by rojopolis** [comment]
> I was also hoping to get some advice regarding testing this. I can set my `sshKeyID` in the Workspace CR and apply it, but it doesn't seem to trigger an update in TFC. I have a named key that I called `testkey` and I was setting it in the CR, but I never saw it show up here in the web console.
> 
> https://i.imgur.com/4k7ZChM.png
> 
> Could you walk me through how you got the ssh key to appear in TFC in your testing? Thanks!
> 
> Here is the field in my CR:
> 
> ```
> $ kubectl describe workspace -n preprod |grep Ssh
>   Ssh Key ID:            testkey
> ```

Oh, I see the problem...  There's a little usability issue with TF Cloud... There is no way to get the SSHKeyID from the web interface.  You're entering the `name` which makes sense since it's the only attribute available.  To get the id you need to list the keys using the API: https://www.terraform.io/docs/cloud/api/ssh-keys.html#list-ssh-keys.  From the response get the `id` field from the key you want to use.  Mine all have the form `sshkey-xxxxx`.  This is the value you should use in the sshKeyID field in the CR.

**Comment by dak1n1** [comment]
Hmm, it might need a rebase since it's not letting me merge.

**Comment by rojopolis** [comment]
> Hmm, it might need a rebase since it's not letting me merge.

Thanks, I updated the branch.

**Pull request merged**